packages/web3-eth-accounts/src/common/utils.ts

```diff
@@ -0,0 +1,619 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import { isHexPrefixed, isHexString } from 'web3-validator';
+import { recoverPublicKey } from 'ethereum-cryptography/secp256k1';
+import { Hardfork } from './enums';
+import {
+	NestedBufferArray,
+	ToBufferInputTypes,
+	TypeOutput,
+	TypeOutputReturnType,
+	NestedUint8Array,
+} from './types';
+
+type ConfigHardfork =
+	// eslint-disable-next-line @typescript-eslint/ban-types
+	| { name: string; block: null; timestamp: number }
+	| { name: string; block: number; timestamp?: number };
+
+/**
+ * Removes '0x' from a given `String` if present
+ * @param str the string value
+ * @returns the string without 0x prefix
+ */
+export const stripHexPrefix = (str: string): string => {
+	if (typeof str !== 'string')
+		throw new Error(`[stripHexPrefix] input must be type 'string', received ${typeof str}`);
+
+	return isHexPrefixed(str) ? str.slice(2) : str;
+};
+/**
+ * Transforms Geth formatted nonce (i.e. hex string) to 8 byte 0x-prefixed string used internally
+ * @param nonce string parsed from the Geth genesis file
+ * @returns nonce as a 0x-prefixed 8 byte string
+ */
+function formatNonce(nonce: string): string {
+	if (!nonce || nonce === '0x0') {
+		return '0x0000000000000000';
+	}
+	if (isHexPrefixed(nonce)) {
+		return `0x${stripHexPrefix(nonce).padStart(16, '0')}`;
+	}
+	return `0x${nonce.padStart(16, '0')}`;
+}
+
+/**
+ * Converts a `Number` into a hex `String`
+ * @param {Number} i
+ * @return {String}
+ */
+const intToHex = function (i: number) {
```

Comment:
Could we replace this with [numberToHex](https://github.com/web3/web3.js/blob/de5617ce4f0f50785e075c7638deeac1d6ed615d/packages/web3-utils/src/converters.ts#L177)?

Reply:
+1 ( see if we can reuse existing functions in our lib )

Reply:
Our `numberToHex` can use negative or decimals numbers but `intToHex` - no

Reply:
IMO, for now we can continue using functions from eth/tx eth/common lib for having stable tx functionality by merging this PR. and after that in near future we can optimize this part by reusing our lib functions and removing unused code.